src/watch.ts

```diff
@@ -79,20 +116,7 @@ export class Watch {
                 // ignore parse errors
             }
         });
-        stream.on('error', error);
-        stream.on('close', done);
 
-        const req = this.requestImpl.webRequest(requestOptions, (err, response, body) => {
```

Comment:
I don't think we want to eliminate this callback from the web request.

Reply:
If "response" event is used together with error event on stream, then you don't need the callback. The stream that is returned by webRequest should emit error event if anything went wrong (i.e. IIRC that's how got behaves https://github.com/sindresorhus/got). So there is one place where the error is handled from watcher's perspective instead of handling errors in stream error event handler and also the callback which leads to ambiguity and is quite difficult to understand how both error handlers relate to each other. More about proper error handling when using request library is here: https://github.com/request/request/issues/647 . The conclusion of that very long discussion (if you scroll all the way down) is basically what I have implemented here.

Reply:
ack. thanks for clarifying.